internal/tablemap.go

```diff
@@ -127,3 +127,43 @@ func GetSpannerCols(conv *Conv, srcTable string, srcCols []string) ([]string, er
 	}
 	return spCols, nil
 }
+
+// GetSpannerKeyName maps source foreign key name to
+// legal Spanner foreign key name
+// If the srcKeyName is empty string we can just return
+// empty string without error.
+// If the srcKeyName is not empty we need to make sure
+// of the following things:
+// a) the new foreign key name is legal
+// b) the new foreign key name doesn't clash with other Spanner
+//    foreignkey names
+// c) we consistently return the same key name.
```

Comment:
I think you should delete part (c).

If I understand correctly, each foreign key constraint name we encounter is unique (if it is non-empty), since we only see foreign key constraint declarations (and not, for example, deletions where you'd see a previously declared constraint name) -- can you confirm?

In other words, what you've implemented is actually what we need.

But that means part (c) doesn't hold: we don't keep around the mapping of constraint names, just the set of names we've used already, so we can't consistently return the same key name.

Maybe worth a comment:

// Note that foreign key names are always unique, so we don't need to consistently
// return the same key name (as we do for table and column mapping).

Reply:
I noticed in mysql that foreign key constraint name is unique across schema, but that doesn't true for postgres, postgres can have duplicate name as long as they are declared in different tables.

so this collision handling is specifically for postgres.

so i can add comment like this:
// Note that foreign key names in postgres can be duplicate in schema, but
// spanner doesn't accept that, so we need to map that source key name to unique 
// spanner key name.